### ISSUE TITLE
Numbering of Search Results In The Reader Interface Is Confusing

### ISSUE THREAD

**Issue by Santoz2** [ManifoldScholar/manifold#2756]
Manifold's Reader Interface Search Results display the total number of results as well as the current number of results being shown. However, it appears that the number of results shown does not update when more results are shown. Further, for non-visual users, it would be helpful if the current position in the Search Results list was read.

## Expected Behavior
Users should only be presented with accurate information. The number of search results shown should update if more results are shown. Reporting to non-visual users their current location in the list of results would also be valuable to the users' experience.

## Current Behavior
The "Showing x-y of z results:" text does not update when more results are shown. The user might guess incorrectly certain features of the Search Results because of this; if the "SHOW ALL EXCERPTS" button does not increase the number of "shown" search results, then a user might guess that there are pages of results. However, there are no pages, so it may cause confusion. 

![image](https://user-images.githubusercontent.com/43553368/83059107-3eb11380-a00e-11ea-915a-98a4bdee2410.png)

## Steps to Reproduce
1. Navigate to the Reader Interface
2. Navigate to a project with readable and searchable text
3. Navigate into a text contained in that project
4. Navigate to and activate the search button in the menu
5. Enter any search criteria into the edit field and submit the search
6. Note that the number of results is reported, for example "Showing 1-20 of 30 results"
7. Navigate to and activate the "SHOW ALL EXCERPTS" button for a section. Notice that, though there are more results shown visibly, the number of results in the "showing x-y of z results" text does not update.

## Testing Environment
NVDA 2020.1
ESR Firefox 68.8.0 esr

**Comment by zdavis** [comment]
@bot bug

**Comment by zdavis** [comment]
Thank you for taking the time to report this bug. We've reviewed the issue and agree that this should be fixed. We’re adding a “planned” label to indicate that we consider this bug as part of our current work plan.

_This was an automated message, but please don't hesitate to reply. Our team watches these issues closely and will respond as soon as we're able to!_

**Comment by zdavis** [comment]
Thanks for submitting this issue!

In this case, I think that we're not going to make the requested change. In the reader, we show two facets in the search results: text sections (aka chapters) and annotations. If I search for a word that appears in 4 chapters, we'll only see 4 chapters in the search results. Now, that word might appear 20 times in each chapter, which is why we show excerpts underneath each result. The list of excerpts is an accordion within a single search result. Opening and closing that accordion doesn't change the number of paginated results.

I am going to create and assign a new issue to @dananjohnson to review whether the implementation of the accordion pattern here is correct. It's entirely possible that we need to improve the a11y here, but given the complexity and constraints around showing these nested search results, I don't see us changing the general approach here.

Previously, we did in fact store every text node as a separate searchable object in the Elasticsearch index. This led to very high indexing times and a lot of overhead, which was causing other problems for our users. We moved these to nested objects in the index, and are displaying them that way to match how they're indexed. The underlying library we use here (searchkick) constrains us in some ways around how we search—breaking the nested excerpts out into a a mixed, paginated result set would require more refactoring and development than we can tackle around this right now.

best,
Zach
